server/eventlog.go

```diff
@@ -475,17 +474,17 @@ func getGrubState(hash crypto.Hash, events []*pb.Event) (*pb.GrubState, error) {
 			if suffixAt == -1 {
 				return nil, fmt.Errorf("invalid prefix seen for PCR%d event: %s", index, rawData)
 			}
-			hasher.Write(rawData[suffixAt : len(rawData)-1])
-			if !bytes.Equal(event.Digest, hasher.Sum(nil)) {
-				// Older GRUBs measure "grub_cmd " with the null terminator.
-				// However, "grub_kernel_cmdline " measurements also ignore the null terminator.
-				hasher.Reset()
-				hasher.Write(rawData[suffixAt:])
-				if !bytes.Equal(event.Digest, hasher.Sum(nil)) {
-					return nil, fmt.Errorf("invalid digest seen for GRUB event log in event %d: %s", idx, hex.EncodeToString(event.Digest))
+
+			if len(rawData) > suffixAt && rawData[len(rawData)-1] == '\x00' {
```

Comment:
why is the len(rawData) > suffixAt check needed? you just need to check it's null-terminated.

Reply:
I want to make sure after the suffix, rawData has at least one byte.

I was thinking if we ever add an empty string as an acceptable suffix, we need to check the length of the rawData to make sure the check is not out of bound.

Reply:
We can cover that with a test. I would prefer to do that over making already complex code harder to read

Reply:
> I changed the it to len(rawData[suffixAt:]) > 0 and add some comment, hopefully it's more clear right now (I'm not sure how to cover this in a test..)

A simple comment or test that validPrefixes are all not "" if your concern is we ever make "" a valid prefix. I really don't think that would ever happen is why I think this check is unnecessary, but I'll leave this up to you.